Sources/Atomic.swift

```diff
@@ -8,27 +8,37 @@
 
 import Foundation
 
-final class PosixThreadMutex: NSLocking {
-	private var mutex = pthread_mutex_t()
+internal struct PosixThreadMutex {
```

Comment:
Why is this a `struct` an `UnsafeMutablePointer` instead of a `class` with a `deinit`?